mlflow/transformers/llm_inference_utils.py

```diff
@@ -106,27 +106,34 @@ def preprocess_llm_inference_params(
             "`data` is expected to be a pandas DataFrame for MLflow inference task after signature "
             f"enforcement, but got type: {type(data)}."
         )
-
-    updated_data = []
-    params = {}
-
-    for column in data.columns:
-        if column in ["prompt", "messages"]:
-            updated_data = data[column].tolist()
-        else:
-            param = data[column].tolist()[0]
-            if column == "max_tokens":
-                params["max_new_tokens"] = param
-            elif column == "stop":
-                source_model_name = (
-                    flavor_config.get("source_model_name") if flavor_config else None
-                )
-                if stop := _get_stopping_criteria(param, source_model_name):
-                    params["stopping_criteria"] = stop
-            else:
-                params[column] = param
-
-    return updated_data, params
+    flavor_config = flavor_config or {}
+    params = params or {}
+    # Pandas convert None to np.nan internally, which is not preferred
+    data = data.replace(np.nan, None).to_dict(orient="list")
```

Comment:
With the reliance on a Pandas API here for casting of invalid data types, I'm wondering if it might be safer to do the conversion on the dictionary instead of the DF? WDYT?

Reply:
That sound reasonable. The initial reason doing this was `replace` allows us to mutate all rows at once like `[0.1, nan, nan] -> [0.1, None, None]`. However, we actually only use the first row anyway, so it's simpler to do that in the dictionary form of the first item.

Reply:
Hmm it is actually a bit tedious to check nan by ourselves, due to inconsistent Pandas null representation - `np.nan`, `float("nan")`, `pd.NA`, each of them is a different object. Lemme use pandas API here to avoid corner case.